src/MatrixDotNet/Extensions/Performance/MultiplyMatrixUnsafe.cs

```diff
@@ -0,0 +1,76 @@
+using System.Threading.Tasks;
+using MatrixDotNet.Exceptions;
+using MatrixDotNet.Extensions.Conversion;
+
+namespace MatrixDotNet.Extensions.Performance
+{
+    public static partial class Optimization
+    {
+        public static async ValueTask<Matrix<T>> MultiplyStrassenAsync<T>(Matrix<T> a, Matrix<T> b)
+            where T : unmanaged
+        {
+            if (a.Rows > 5000)
+            {
+                throw new MatrixDotNetException("Matrix is much big size your CPU will be on 100%");
+            }
+
+            return await MultiplyStrassenParallel(a, b);
+        }
+
+        private static ValueTask<Matrix<T>> MultiplyStrassenParallel<T>(Matrix<T> a, Matrix<T> b)
+            where T : unmanaged
+        {
+            if (a.Rows <= 64)
+            {
+                return new ValueTask<Matrix<T>>(a * b);
+            }
+
+            a.SplitMatrix(out var a11, out var a12, out var a21, out var a22);
+            b.SplitMatrix(out var b11, out var b12, out var b21, out var b22);
+
+            var p1 = Task.Run(() => MultiplyStrassen(a11 + a22, b11 + b22));
+            var p2 = Task.Run(() => MultiplyStrassen(a21 + a22, b11));
+            var p3 = Task.Run(() => MultiplyStrassen(a11, b12 - b22));
+            var p4 = Task.Run(() => MultiplyStrassen(a22, b21 - b11));
+            var p5 = Task.Run(() => MultiplyStrassen(a11 + a12, b22));
+            var p6 = Task.Run(() => MultiplyStrassen(a21 - a11, b11 + b12));
+            var p7 = Task.Run(() => MultiplyStrassen(a12 - a22, b21 + b22));
```

Comment:
What do you think about arrays?

Reply:
I think it is will be less readable and there will be no effect in performance

Reply:
but we used params in `Task.WaitAll` which creates implicit array